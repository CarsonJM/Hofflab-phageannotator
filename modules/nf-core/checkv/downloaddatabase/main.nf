process CHECKV_DOWNLOADDATABASE {
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/checkv:1.0.1--pyhdfd78af_0':
        'biocontainers/checkv:1.0.1--pyhdfd78af_0' }"

    output:
    path "${prefix}/"           , emit: checkv_db
    path "${prefix}/README.txt" , emit: readme
    path "${prefix}/**.tsv"     , emit: tsv
    path "${prefix}/**.faa"     , emit: faa
    path "${prefix}/**.fna"     , emit: fna
    path "${prefix}/**.hmm"     , emit: hmm
    path "${prefix}/**.dmnd"    , emit: dmnd
    path "versions.yml"         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args        = task.ext.args ?: ''
    prefix          = task.ext.prefix ?: "checkv_db"

    """
    checkv download_database \\
        download \\
        ${args}

    mv download/* ${prefix}/

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        checkv: \$(checkv -h 2>&1  | sed -n 's/^.*CheckV v//; s/: assessing.*//; 1p')
    END_VERSIONS
    """

    stub:
    def args    = task.ext.args ?: ''
    prefix      = task.ext.prefix ?: "checkv_db"

    """
    mkdir -p ${prefix}/genome_db
    touch ${prefix}/README.txt
    touch ${prefix}/genome_db/changelog.tsv
    touch ${prefix}/genome_db/checkv_error.tsv
    touch ${prefix}/genome_db/checkv_info.tsv
    touch ${prefix}/genome_db/checkv_reps.faa
    touch ${prefix}/genome_db/checkv_reps.fna
    touch ${prefix}/genome_db/checkv_reps.tsv
    touch ${prefix}/genome_db/checkv_reps.dmnd
    mkdir -p ${prefix}/hmm_db/
    touch ${prefix}/hmm_db/test.hmm
    touch ${prefix}/hmm_db/checkv_hmms.tsv
    touch ${prefix}/hmm_db/genome_lengths.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        checkv: \$(checkv -h 2>&1  | sed -n 's/^.*CheckV v//; s/: assessing.*//; 1p')
    END_VERSIONS
    """
}
