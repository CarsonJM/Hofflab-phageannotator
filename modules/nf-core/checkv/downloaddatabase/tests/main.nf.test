nextflow_process {

    name "Test Process CHECKV_DOWNLOADDATABASE"
    script "../main.nf"
    process "CHECKV_DOWNLOADDATABASE"


    test("No input: download only") {
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.readme,
                    process.out.tsv,
                    process.out.faa,
                    file(process.out.dmnd[0]).name,
                    file(process.out.hmm[0].find { file(it).name ==~ /.*\.hmm/ }).name,
                    file(process.out.fna.find { file(it).name ==~ /.*\.fna/ }).name,
                    process.out.versions
                    ).match()
                }
            )
        }
    }


    test("No input: download only - stub") {
        options "-stub"
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
