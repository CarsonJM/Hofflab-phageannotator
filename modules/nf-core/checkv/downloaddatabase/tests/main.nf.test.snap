{
    "No input: download only - stub": {
        "content": [
            {
                "0": [
                    [
                        "README.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                        [
                            "changelog.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "checkv_error.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "checkv_info.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "checkv_reps.dmnd:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "checkv_reps.faa:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "checkv_reps.fna:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "checkv_reps.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ],
                        [
                            "checkv_hmms.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "genome_lengths.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.hmm:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "README.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    [
                        "changelog.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "checkv_error.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "checkv_info.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "checkv_reps.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "checkv_hmms.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "genome_lengths.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "checkv_reps.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "4": [
                    "checkv_reps.fna:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "5": [
                    "test.hmm:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "6": [
                    "checkv_reps.dmnd:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "7": [
                    "versions.yml:md5,29c4fbe7a6cb8b6c0913363d718cce9f"
                ],
                "checkv_db": [
                    [
                        "README.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                        [
                            "changelog.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "checkv_error.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "checkv_info.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "checkv_reps.dmnd:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "checkv_reps.faa:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "checkv_reps.fna:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "checkv_reps.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ],
                        [
                            "checkv_hmms.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "genome_lengths.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.hmm:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "dmnd": [
                    "checkv_reps.dmnd:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "faa": [
                    "checkv_reps.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "fna": [
                    "checkv_reps.fna:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "hmm": [
                    "test.hmm:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "readme": [
                    "README.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "tsv": [
                    [
                        "changelog.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "checkv_error.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "checkv_info.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "checkv_reps.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "checkv_hmms.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "genome_lengths.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,29c4fbe7a6cb8b6c0913363d718cce9f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-24T14:19:33.975893953"
    },
    "No input: download only": {
        "content": [
            [
                "README.txt:md5,1aa190eec91df1b347ee7c6551924ad5"
            ],
            [
                [
                    "changelog.tsv:md5,b10bfc8960726fb6f17040d03cc71ded",
                    "checkv_error.tsv:md5,755309dc403230b46794b2b211a67036",
                    "checkv_info.tsv:md5,8657804d8ef1a1ef9765d0978c8933f3",
                    "checkv_reps.tsv:md5,8868dfc509af82daf0eeccd2079d9873",
                    "checkv_hmms.tsv:md5,8ca7392ff5e911332b0a9fec894839bf",
                    "genome_lengths.tsv:md5,e88708112a126a6b0ba441c44c080ff6"
                ]
            ],
            [
                "checkv_reps.faa:md5,253dc981de41db728167b9ad95834a52"
            ],
            "checkv_reps.dmnd",
            "1.hmm",
            "checkv_reps.fna",
            [
                "versions.yml:md5,29c4fbe7a6cb8b6c0913363d718cce9f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-24T14:18:53.225948142"
    }
}